### DISCUSSION TITLE
how to deploy as pwa?

### DISCUSSION THREAD

**Discussion by wideweide** [ONLYOFFICE/DocumentServer#1825] [Q&A]
**Do you want to request a *feature* or report a *bug*?**

feature

**What is the current behavior?**
It is slow when first even after visit onlyoffice

**What is the expected behavior?**
use service worker to cache html/css/js,then only first time visit is slow,then very fast

**Did this work in previous versions of DocumentServer?**

I am not sure,but I searched github and site, not to see ‘pwa’ or service workder

**Comment by ShockwaveNN** [comment]
Hi, I don't think it's possible in our architecture

**Comment by kxu1988** [comment]
> **Do you want to request a _feature_ or report a _bug_?**
> 
> feature
> 
> **What is the current behavior?** It is slow when first even after visit onlyoffice
> 
> **What is the expected behavior?** use service worker to cache html/css/js,then only first time visit is slow,then very fast
> 
> **Did this work in previous versions of DocumentServer?**
> 
> I am not sure,but I searched github and site, not to see ‘pwa’ or service workder

this is  a good idea, maybe we can try
